packages/mesos-client/src/createEventStreamConnection.js

```diff
@@ -0,0 +1,34 @@
+import Rx from "rxjs";
+import { read } from "recordio";
+import { XHRConnection } from "connection-manager";
+
+import observableFromXhrConnection from "./observableFromXhrConnection";
+
+function extractRecords({ buffer = "", position = 0 } = {}, data = "") {
```

Comment:
This will need tests and as such we need to think about how we make it testable.

Reply:
Yes and no, the read function is covered in the `recordio` package

Reply:
I think it still makes sense to test this function! How about separating all of the transformation functions into a separate file(s) so that we can properly test them?